art/attacks/evasion/hop_skip_jump.py

```diff
@@ -158,25 +175,54 @@ def generate(self, x: np.ndarray, y: Optional[np.ndarray] = None, **kwargs) -> n
         for ind, val in enumerate(tqdm(x_adv, desc="HopSkipJump", disable=not self.verbose)):
             self.curr_iter = start
             if self.targeted:
-                x_adv[ind] = self._perturb(
-                    x=val,
-                    y=y[ind],
-                    y_p=preds[ind],
-                    init_pred=init_preds[ind],
-                    adv_init=x_adv_init[ind],
-                    clip_min=clip_min,
-                    clip_max=clip_max,
-                )
+                if mask is not None:
+                    x_adv[ind] = self._perturb(
+                        x=val,
+                        y=y[ind],
+                        y_p=preds[ind],
+                        init_pred=init_preds[ind],
+                        adv_init=x_adv_init[ind],
+                        mask=mask[ind],
+                        clip_min=clip_min,
+                        clip_max=clip_max,
+                    )
+
+                else:
+                    x_adv[ind] = self._perturb(
+                        x=val,
+                        y=y[ind],
+                        y_p=preds[ind],
+                        init_pred=init_preds[ind],
+                        adv_init=x_adv_init[ind],
+                        mask=None,
+                        clip_min=clip_min,
+                        clip_max=clip_max,
+                    )
+
             else:
-                x_adv[ind] = self._perturb(
-                    x=val,
-                    y=-1,
-                    y_p=preds[ind],
-                    init_pred=init_preds[ind],
-                    adv_init=x_adv_init[ind],
-                    clip_min=clip_min,
-                    clip_max=clip_max,
-                )
+                if mask is not None:
+                    x_adv[ind] = self._perturb(
+                        x=val,
+                        y=-1,
+                        y_p=preds[ind],
+                        init_pred=init_preds[ind],
+                        adv_init=x_adv_init[ind],
+                        mask=mask[ind],
+                        clip_min=clip_min,
+                        clip_max=clip_max,
+                    )
+
+                else:
+                    x_adv[ind] = self._perturb(
+                        x=val,
+                        y=-1,
+                        y_p=preds[ind],
+                        init_pred=init_preds[ind],
+                        adv_init=x_adv_init[ind],
+                        mask=None,
+                        clip_min=clip_min,
+                        clip_max=clip_max,
+                    )
```

Comment:
Would it be possible to change the value of `mask` so that we only need one function call? E.g. make mask a list of None if no mask is provided?

Reply:
Solved.